pkg/data/domain/app/service.go

```diff
@@ -37,40 +37,80 @@ func New(config Config) (Interface, error) {
 	return s, nil
 }
 
-// Get fetches an App CR by name and namespace.
-func (s *Service) Get(ctx context.Context, options GetOptions) (*applicationv1alpha1.App, error) {
-	namespacedName := types.NamespacedName{
-		Namespace: options.Namespace,
-		Name:      options.Name,
+// Get fetches a list of app CRs filtered by namespace and optionally by
+// name.
+func (s *Service) Get(ctx context.Context, options GetOptions) (Resource, error) {
+	var resource Resource
+	var err error
+
+	if len(options.Name) > 0 {
+		resource, err = s.getByName(ctx, options.Name, options.Namespace)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	} else {
```

Comment:
I don't think we need `else` statement here, right?

Reply:
Good catch we don't need it.